singer_sdk/mapper.py

```diff
@@ -831,3 +832,38 @@ def register_raw_stream_schema(  # noqa: PLR0912, C901
             else:
                 # Additional mappers for aliasing and multi-projection:
                 self.stream_maps[source_stream].append(mapper)
+
+    @staticmethod
+    def _eval_stream(expr: str, stream_name: str) -> str:
+        """Solve an alias expression.
+
+        Args:
+            expr: String expression to evaluate.
+            stream_name: Name of stream to transform.
+
+        Returns:
+            Evaluated expression.
+
+        Raises:
+            MapExpressionError: If the mapping expression failed to evaluate.
+        """
+        # Allow stream name access within alias transform
+        names = {"__stream_name__": stream_name}
+
+        try:
+            expr_evaluator = simpleeval.EvalWithCompoundTypes(names=names)
+            result: str = expr_evaluator.eval(expr)
+        except simpleeval.NameNotDefined:
+            logging.debug(
+                "Failed to evaluate simpleeval expression %(expr) - "
+                "falling back to original expression",
+                extra={"expr": expr},
+            )
+            result: str = expr
```

Comment:
I am swallowing NameNotDefined errors here to support existing `__alias__` configurations. Since `__alias__` was not previously evaluated, it contains plain strings - which the evaluator tries to read as python objects.

For instance, `{ "__alias__": "aliased_stream" }` will produce `NameNotDefined`. It would need single quotes to avoid this, like `{ "__alias__": "'aliased_stream'" }`, which is a breaking change.

Please let me know what you think

Reply:
I think this makes sense. Thanks for being mindful of breaking changes 🙂